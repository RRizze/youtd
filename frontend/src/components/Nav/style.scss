.nav {
  display: flex;
  justify-content: center;
  position: relative;
  padding-left: 10px;
  .list {
    margin: 0 auto;
    padding: 0;
  }
  .list-item {
    display: inline-block;
    margin-right: 10px;
  }
}