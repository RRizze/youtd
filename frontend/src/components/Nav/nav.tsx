import React from "react";
import NavItem from "../NavItem";

import "./style.scss";

interface NavItem {
  to: string;
  name: string;
}

interface Props {
  items: NavItem[];
}

const Nav = ({ items }: Props) => (
  <nav className="nav">
    <ul className="list">
      {items.map((item) => {
        return (
          <li
            key={item.name}
            className="list-item"
          >
            <NavItem to={item.to}>{item.name}</NavItem>
          </li>
        );
      })}
    </ul>
  </nav>
);

export default Nav;
