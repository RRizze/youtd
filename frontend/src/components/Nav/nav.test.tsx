import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Nav from "./nav";

describe("Nav tests", () => {
  it("render correctly", () => {
    const tree = renderer
      .create(<Nav items={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render with props, it has some nav-items", () => {
    const props = {
      items: [
        { to: "/about", name: "about" },
        { to: "/towers", name: "towers" },
        { to: "/items", name: "items" },
      ],
    },
      wrapper = shallow(<Nav {...props} />),
      items = wrapper.find("li");

    expect(items.length).toBe(3);
  });
});
