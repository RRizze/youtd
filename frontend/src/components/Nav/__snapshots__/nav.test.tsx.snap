// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav tests render correctly 1`] = `
<nav
  className="nav"
>
  <ul
    className="list"
  />
</nav>
`;
